bson/datetime_ms.py

```diff
@@ -127,24 +129,33 @@ def _millis_to_datetime(millis: int, opts: CodecOptions) -> Union[datetime.datet
         or opts.datetime_conversion == DatetimeConversion.DATETIME_CLAMP
         or opts.datetime_conversion == DatetimeConversion.DATETIME_AUTO
     ):
-        tz = opts.tzinfo or datetime.timezone.utc
-        if opts.datetime_conversion == DatetimeConversion.DATETIME_CLAMP:
-            millis = max(_min_datetime_ms(tz), min(millis, _max_datetime_ms(tz)))
-        elif opts.datetime_conversion == DatetimeConversion.DATETIME_AUTO:
-            if not (_min_datetime_ms(tz) <= millis <= _max_datetime_ms(tz)):
-                return DatetimeMS(millis)
-
-        diff = ((millis % 1000) + 1000) % 1000
-        seconds = (millis - diff) // 1000
-        micros = diff * 1000
-
-        if opts.tz_aware:
-            dt = EPOCH_AWARE + datetime.timedelta(seconds=seconds, microseconds=micros)
-            if opts.tzinfo:
-                dt = dt.astimezone(tz)
-            return dt
-        else:
-            return EPOCH_NAIVE + datetime.timedelta(seconds=seconds, microseconds=micros)
+        try:
+            tz = opts.tzinfo or datetime.timezone.utc
```

Comment:
Perhaps put this part in a function to avoid such a large `try:` section.

Reply:
I think we can move the "try:" down to cover only the lines that might raise the OverflowError.

Reply:
Yeah, I'll go ahead and move this down to 144. (Right before the first arithmetic operation occurs)